docs/guide/core-concepts.rst

```diff
@@ -13,364 +13,489 @@ alike. If GIS is brand, brand new to you, `this
 document <https://www.gislounge.com/what-is-gis/>`__ is a useful high
 level overview.
 
-Basic Terms
-===========
+Glossary
+========
 
--  **Tile:** A grid of numeric *cells* that represent some data on the
-   Earth.
--  **Cell:** A single unit of data in some grid, also called a
-   *Location* in GIS.
--  **Layer:** or "Tile Layer", this is a grid (or cube) of *Tiles*.
--  **Zoom Layer:** a *Tile Layer* at some zoom level.
--  **Key:** Used to index a *Tile* in a grid (or cube) of them.
--  **Key Index:** Used to transform higher-dimensional *Keys* into one
-   dimension.
--  **Metadata:** or "Layer Metadata", stores information critical to
-   Tile Layer IO.
--  **Layout Definition:** A description of a Tile grid (its dimensions,
-   etc).
--  **Extent:** or "Bounding Box", represents some area on the Earth.
--  **Raster:** A *Tile* with an *Extent*.
--  **Vector:** or "Geometry", these are Point, Line, and Polygon data.
--  **Feature:** A *Geometry* with some associated metadata.
--  **RDD:** "Resilient Distributed Datasets" from `Apache
-   Spark <http://spark.apache.org/>`__. Can be thought of as a highly
-   distributed Scala ``Seq``.
+- `Vector <#geometries>`_ or **Geometry**: Structures built up by
+  connecting Points in space; includes ``Point``\s, ``Line``\s,
+  ``Polygon``\s.
+- `Extent <#extents>`_ or **Bounding Box**: An axis aligned, rectangular
+  region.
+- `Feature <#features>`_: A *Geometry* with some associated metadata.
+- `Cell <#working-with-cell-values>`_: A single unit of data in some
+  grid.
+- `Tile <#tiles-and-rasters>`_: A grid of numeric *cells* that represent
+  some data on the Earth.
+- `Raster <#rasters>`_: A *Tile* with an *Extent*.
+- **RDD:** "Resilient Distributed Datasets" from `Apache Spark <http://spark.apache.org/>`__. Can be thought of as a distributed Scala ``Seq``.
+- `Key <#keys>`_: Used to index a *Tile* in a grid (or cube) of them.
+- `Key Index <#key-indexes>`_: A means to transform higher-dimensional
+  *Keys* into a unique one-dimensional identifier.
+- `Layer <#layouts-and-tile-layers>`_ or **Tile Layer**: This is a grid
+  (or cube) of *Tiles*.
+- `Metadata <#layouts-and-tile-layers>`_ or **Layer Metadata**: A
+  descriptive structure that defines how to interpret a key-value store as a
+  *Layer*.
+- `Layout Definition <#layouts-and-tile-layers>`_: A structure that
+  relates *Keys* to geographic locations and vice versa.
 
```

Comment:
Having `Vector Tiles` and `Type classes` listed in the glossary should be done as well. I'm not sure if the `GeoTiff` section needs to be added, but it wouldn't hurt.

Reply:
I think we should recall that this is a glossary and not a TOC.  The TOC will be in the sidebar on readthedocs.  I want to minimize the load of terms, so I'm restricting the glossary to ideas that are really fundamental to GT, and you couldn't just google for them.  Vector tiles are well-known, and easily discovered on the web.  I'd argue that, if anything, they aren't "core concepts", and maybe shouldn't be covered in these docs at all.

Type classes shouldn't be in the glossary because, again, it's a glossary, not a TOC.  The common GT typeclasses are not fundamental to understanding the system, and pretty inside baseball.  They could be pulled out into a separate doc, but this is probably as good a place for them as any.

GeoTiff could be in the glossary because of its importance, though it is an easily discoverable idea not endemic to GT.

I'm aiming to really put a cap on the amount of terms in the glossary.  I'd say the Key Index isn't that important and could actually be removed.